tests/rptest/tests/partition_movement_test.py

```diff
@@ -368,7 +368,19 @@ def test_bootstrapping_after_move(self, num_to_upgrade):
 
         # snapshot offsets
         rpk = RpkTool(self.redpanda)
-        partitions = rpk.describe_topic(spec.name)
+
+        def has_offsets_for_all_partitions(out):
+            # NOTE: partitions may not be returned if their fields can't be
+            # populated, e.g. during leadership changes.
+            partitions = list(rpk.describe_topic(spec.name))
+            if len(partitions) == 3:
+                out += partitions
+                return True
+            return False
+
+        partitions = []
+        wait_until(lambda: has_offsets_for_all_partitions(partitions), 30, 2)
```

Comment:
i think you can use `tests/rptest/util.py:wait_until_result` to get info out of the wait_until predicate

Reply:
Nice!